tests/test_preprints.py

```diff
@@ -117,3 +118,19 @@ def test_non_admin_update_file(self):
         with assert_raises(PermissionsError):
             self.project.set_preprint_file(self.file_two, auth=self.read_write_user_auth, save=True)
         assert_equal(self.project.preprint_file._id, self.file._id)
+
+
+class TestPreprintProviders(OsfTestCase):
+    def setUp(self):
+        super(TestPreprintProviders, self).setUp()
+        self.preprint = PreprintFactory(provider=None)
+
+    def test_set_provider(self):
+        new_provider = PreprintProviderFactory(name='WWEArxiv')
+
+        assert_equal(self.preprint.preprint_provider, None)
+
+        self.preprint.preprint_provider = new_provider
```

Comment:
This field name changed. Also, I think it should be calling `add_preprint_provider(new_provider, self.user, save=True)`. Please add a test for `remove_preprint_provider` too